tests/test_models.py

```diff
@@ -864,6 +867,120 @@ def test_string_conversion(self, method):
 
 
 @mark.django_db
+class TestDataSharingConsentManager(unittest.TestCase):
+    """
+    Tests for the custom Data Sharing Consent Manager.
+    """
+
+    def setUp(self):
+        super(TestDataSharingConsentManager, self).setUp()
+        DataSharingConsentFactory(
+            enterprise_customer=EnterpriseCustomerFactory(
+                name='rich_enterprise'
+            ),
+            username='lowly_bob',
+            course_id='hard_course_2017'
+        )
+
+    def test_get_returns_proxy_when_consent_doesnt_exist(self):
+        """
+        Test that ``get`` on custom manager returns a ``ProxyDataSharingConsent`` object when
+        the searched-for ``DataSharingConsent`` object doesn't exist.
+        """
+        dsc = DataSharingConsent.objects.get(username='lowly_bob')
+        proxy_dsc = DataSharingConsent.objects.get(username='optimistic_bob')
+        self.assertIsInstance(dsc, DataSharingConsent)
+        self.assertIsInstance(proxy_dsc, ProxyDataSharingConsent)
+        self.assertNotEqual(dsc, proxy_dsc)
+
+    def test_get_returns_consent_when_it_exists(self):
+        """
+        Test that ``get`` on custom manager returns a ``DataSharingConsent`` object when the searched-for
+        ``DataSharingConsent`` object exists.
+        """
+        dsc = DataSharingConsent.objects.get(username='lowly_bob')
+        same_dsc = DataSharingConsent.objects.get(username='lowly_bob')
+        self.assertIsInstance(same_dsc, DataSharingConsent)
+        self.assertEqual(dsc, same_dsc)
+
+
+@ddt.ddt
+class TestProxyDataSharingConsent(TransactionTestCase):
+    """
+    Tests of the ``ProxyDataSharingConsent`` class (pseudo-model).
+    """
+
+    def setUp(self):
+        super(TestProxyDataSharingConsent, self).setUp()
+        self.proxy_dsc = ProxyDataSharingConsent(
+            enterprise_customer=EnterpriseCustomerFactory(
+                name='rich_enterprise'
+            ),
+            username='lowly_bob',
+            course_id='hard_course_2017'
+        )
+
+    def test_commit(self):
+        """
+        Test that ``ProxyDataSharingConsent``'s ``commit`` method properly creates/saves/returns a new
+        ``DataSharingConsent`` instance, or returns ``None`` for any validation errors (i.e. conflict).
+        """
+        new_dsc = self.proxy_dsc.commit()
+        no_dsc = self.proxy_dsc.commit()
+        dsc = DataSharingConsent.objects.all()
+        self.assertEqual(len(dsc), 1)
+        self.assertEqual(dsc[0], new_dsc)
+        self.assertIsNone(no_dsc)
+
+    @ddt.data(
+        str, repr
+    )
+    def test_string_conversion(self, method):
+        """
+        Test ``ProxyDataSharingConsent`` conversion to string
+        """
+        expected_to_str = (
```

Comment:
Given this is an expectation, and that we're only testing with the one set of values, I'd prefer writing out the raw string literal.